### PR TITLE
[GLT-4188] Update HibernateRunDao to JPA Criteria

### PR DESCRIPTION
**Pull request by sinry-d**
Jira ticket or GitHub issue: https://jira.oicr.on.ca/browse/GLT-4188

- [ ] Includes a change file

### CODE REVIEW

**Review by avarsava — APPROVED** [review]

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java [hunk 1]
```diff
@@ -371,29 +358,34 @@ public String getFriendlyName() {
...
+    Join<ListLibraryAliquotView, ParentLibrary> library =
+        builder.getJoin(aliquot, ListLibraryAliquotView_.parentLibrary);
+    Join<ParentLibrary, ParentSample> sample = builder.getJoin(library, ParentLibrary_.parentSample);
+    Join<ParentSample, ParentProject> project = builder.getJoin(sample, ParentSample_.parentProject);
+
+    In<Long> inClause = builder.getCriteriaBuilder().in(builder.getRoot().get(Run_.runId));
+    List<Long> ids = runs.stream().map(Run::getId).toList();
+    for (Long id : ids) {
+      inClause.value(id);
+    }
```
Comment (djcooke) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java:383]: Can avoid creating the `List<Long>` by looping through the runs and adding value `run.getId()` instead

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java [hunk 2]
```diff
@@ -371,29 +358,34 @@ public String getFriendlyName() {
...
+    builder.addPredicate(inClause);
+    builder.setColumns(builder.getRoot().get(Run_.runId), project.get(ParentProject_.code),
+        project.get(ParentProject_.name));
+
+    List<Object[]> results = builder.getResultList();
```
Comment (djcooke) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java:388]: If there are multiple libraries from the same project, is this adding the same project multiple times? I think either this query needs to return distinct results only, or below needs to consider that they are not distinct

I have not deployed this to confirm, but you can check on the Runs list page to see the projects listed for each run

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java (hunk 1 above)
Comment (sinry-d) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java]: Left `List<Long>` in due to the parameters `addInPredicate` takes in QueryBuilder

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java (hunk 2 above)
Comment (sinry-d) [reply sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java:367]: Screenshot from 2024-06-28 08-48-47

**Review by djcooke — APPROVED** [review]

**Pull request merged**